flytekit/core/workflow.py

```diff
@@ -88,11 +88,12 @@ class WorkflowMetadataDefaults(object):
     WorkflowMetadata represents metadata about the workflow itself.
     """
 
-    interruptible: bool
+    interruptible: Optional[bool] = None
 
     def __post_init__(self):
-        if self.interruptible is not True and self.interruptible is not False:
```

Comment:
Yah I think this we want to keep as this strict check because of the IDL link earlier.  According to @katrogan - the reason the `WorkflowMetadataDefaults`'s `interruptible` is not a `oneof` but `NodeMetadata`'s `interruptible` is is because the former may or may not be set but `NodeMetadata` one should indicate whether it's set locally, or should inherit.  Katrina can clarify more if that's incomplete.

Reply:
This makes sense to me!